pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -1691,7 +1705,10 @@ async def _map_messages(  # noqa: C901
                         # If `send_item_ids` is false, we won't send the `BuiltinToolReturnPart`, but OpenAI does not have a type for files from the assistant.
                         pass
                     elif isinstance(item, ThinkingPart):
-                        if item.id and send_item_ids:
+                        # Get raw CoT content from provider_details if present
+                        raw_content: list[str] | None = (item.provider_details or {}).get('raw_content')
```

Comment:
We should only ever read `provider_details` if the `provider_name` matches our `self.system`